### PR TITLE
docs: v0.4 release notes

### PR DESCRIPTION
**Pull request by chappjc**
Rendered: https://github.com/decred/dcrdex/blob/7cb08b8a5d0a6864ccc18bbe05b2b1afec0b68d7/docs/release-notes/release-notes-0.4.0.md

### CODE REVIEW

File: docs/release-notes/release-notes-0.4.0.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
+# DCRDEX v0.4.0
+
+Dec 23, 2021
```
Comment (chappjc) [inline docs/release-notes/release-notes-0.4.0.md:3]: date of the `v0.4.0-rc1` tag.  I suppose we'll update this date as we get to final.

File: docs/release-notes/release-notes-0.4.0.md [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+  wallet types. The `client/asset.Driver` now has `Create` and `Exists` methods
+  for "seeded" (native, built-in) wallet types. `(*Core).ReconfigureWallet`
+  replaces the `assetID` and `cfg` fields with a `core.WalletForm struct`, which
+  now has a `Type string` field. The `core.WalletState` struct now has a new
+  `WalletType string` field. A type of `""` (empty string) will be recognized as
+  the legacy RPC wallets
+  ([d202f6d](https://github.com/decred/dcrdex/commit/d202f6d4d0de6884be7dc1bd73a0e865835d6a9b))
+- (Breaking) Move the shutdown prompt function from client/core into client/cmd/dexc. The
+  `(*Core).PromptShutdown` method is removed.
+  ([1248eb9](https://github.com/decred/dcrdex/commit/1248eb92275cf7dddd98b3cc7c1b97a81be4ab5f)
```
Comment (JoeGruffins) [inline docs/release-notes/release-notes-0.4.0.md:388]: Missing a `)` on the right.

**Review by JoeGruffins — APPROVED** [review]
Looks great! Can also add @martonp to the `AUTHORS` file?

File: docs/release-notes/release-notes-0.4.0.md [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+- The client now uses a seed and deterministic (HD) account key derivation. An
+  application seed is generated during client initialization. When a new dex
+  account is created, the server's identity is used to deterministically
+  generate a client identity for the account. **Users should save a copy of
+  their app seed.** When upgrading, a new seed will be generated, which may be
+  accessed for backup from the Settings page. When initializing a new dex
+  client, there is an option to provide an existing seed. The RPC client
+  (dexcctl) also has an `appseed` request endpoint. If an account is suspended,
+  a new one will be generated from a different derivation path, and on restore
+  from seed, non-suspended accounts will be located.
```
[the remark this answers is not in the document]
Comment (chappjc) [reply docs/release-notes/release-notes-0.4.0.md:141]: Please see what you think with https://github.com/decred/dcrdex/pull/1387/commits/7cb08b8a5d0a6864ccc18bbe05b2b1afec0b68d7

**Review by JoeGruffins — APPROVED** [review]

**Pull request merged**